spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/config/ConfigMapCache.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2013-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.kubernetes.commons.config;
+
+/**
+ * @author wind57
+ */
+public interface ConfigMapCache {
+
+	/**
+	 * Discards all stored entries from the cache.
+	 */
+	void discardAll();
+
+	/**
+	 * an implementation that does nothing. In the next major release it will become
+	 * absolute and must be removed.
+	 */
+	class NOOPCache implements ConfigMapCache {
```

Comment:
this is needed in case someone uses the public methods that we can't alter anymore until the next major release. in their case, the cache will still kick in, but no clean-up will be performed.